Develop

Tomokou-ne opened the pull request:
Задание к 4-й неделе

Tomokou-ne committed:
first commit

Tomokou-ne committed:
Изменены коммиты

Tomokou-ne committed:
Добавлен класс WebSecurityConfig

Tomokou-ne committed:
Добавлены PreAuthorize аннотации

dmitry-bizin commented on logs/app_log.log:
логи надо игнорировать и не коммитить в репозиторий публичный

dmitry-bizin commented on logs/archive/app_log.2023-10-15.0.log:
логи надо игнорировать и не коммитить в репозиторий публичный

dmitry-bizin commented on logs/archive/app_log.2023-10-16.0.log:
логи надо игнорировать и не коммитить в репозиторий публичный

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -55,9 +66,10 @@ public ResponseEntity<ResponseWrapperAds> getAllAds() {
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -72,34 +84,35 @@ public ResponseEntity<Ads> addAd(@RequestPart CreateAds properties, @RequestPart
     )
-    public ResponseEntity<FullAds> getAds(@PathVariable int id) {
-        return ResponseEntity.status(200).build();
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -72,34 +84,35 @@ public ResponseEntity<Ads> addAd(@RequestPart CreateAds properties, @RequestPart
-    @PreAuthorize("@adServiceImpl.getEntity(#id).author.email.equals(#auth.name) or hasAuthority('DELETE_ANY_AD')")
-    public ResponseEntity<?> removeAd(@PathVariable int id, Authentication auth) {
-        return ResponseEntity.status(200).build();
+    @PreAuthorize("@adServiceImpl.getEntity(#id).author.email.equals(#auth.name)")
```
нужна ещё проверка на админа, т.к. админ может тоже вызывать этот метод

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -72,34 +84,35 @@ public ResponseEntity<Ads> addAd(@RequestPart CreateAds properties, @RequestPart
     }
 
     @GetMapping("/me")
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -72,34 +84,35 @@ public ResponseEntity<Ads> addAd(@RequestPart CreateAds properties, @RequestPart
     @PatchMapping(value = "/{id}/image", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
-    public ResponseEntity<?> updateImage(@PathVariable int id, @RequestPart MultipartFile image) {
-        return null;
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

BizinMitya commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -72,34 +84,35 @@ public ResponseEntity<Ads> addAd(@RequestPart CreateAds properties, @RequestPart
     }
 
     @PatchMapping("/{id}")
     @PreAuthorize("@adServiceImpl.getEntity(#id).author.email.equals(#auth.name) or hasAuthority('UPDATE_ANY_AD')")
```
нужна ещё проверка на админа, т.к. админ может тоже вызывать этот метод

dmitry-bizin replied:
где определено authority UPDATE_ANY_AD ?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -34,9 +42,10 @@ public class CommentController {
                     @ApiResponse(responseCode = "404", description = "Not Found")
             }
     )
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -54,9 +63,12 @@ public ResponseEntity<Comment> getComments(@PathVariable("id") Integer id,
-    public ResponseEntity<Comment> addComments(@PathVariable("id") Integer id,
-                                               @NotNull @RequestBody Comment comment) {
-        return ResponseEntity.ok(comment);
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -71,9 +83,11 @@ public ResponseEntity<Comment> addComments(@PathVariable("id") Integer id,
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
+    @PreAuthorize("hasRole('ADMIN')")
```
нужна ещё проверка на хозяина

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -90,10 +104,11 @@ public ResponseEntity<Void> deleteComments(@PathVariable("adId") Integer adId,
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
здесь неверно, поменять может хозяин ИЛИ админ

dmitry-bizin commented on src/main/java/ru/skypro/homework/WebSecurityConfig.java:
```diff
@@ -0,0 +1,50 @@
+                                authorization
+                                        .mvcMatchers(AUTH_WHITELIST)
+                                        .permitAll()
+                                        .mvcMatchers(HttpMethod.GET, "/ads", "/ads/image/**", "/users/image/**")
```
зачем /ads/image/** и /users/image/** ?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -34,8 +44,14 @@ public class UserController {
     )
-    public ResponseEntity<NewPassword> setPassword(@RequestBody NewPassword newPassword) {
-        return ResponseEntity.ok(newPassword);
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -52,8 +68,9 @@ public ResponseEntity<NewPassword> setPassword(@RequestBody NewPassword newPassw
     )
-    public ResponseEntity<User> getUser() {
-        return ResponseEntity.ok(new User());
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -71,8 +88,9 @@ public ResponseEntity<User> getUser() {
     )
-    public ResponseEntity<User> updateUser(@RequestBody User user) {
-        return ResponseEntity.ok(user);
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -85,7 +103,9 @@ public ResponseEntity<User> updateUser(@RequestBody User user) {
     )
-    public ResponseEntity<Void> updateUserImage(@RequestParam MultipartFile image) {
-        return ResponseEntity.status(200).build();
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
```
аннотация не имеет смысла, т.к. ролей и так 2 и обоим разрешён доступ, можно просто удалить аннотацию и ничего не поменяется

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -54,12 +55,12 @@ public Ads update(int id, CreateAds ads) {
-    private AdEntity getEntityById(int id) {
-        return adRepository.findById(id).orElseThrow(() -> new FindNoEntityException("объявление"));
+    private AdEntity getEntityById(Long id) {
+        return adRepository.findById(Math.toIntExact(id)).orElseThrow(() -> new FindNoEntityException("объявление"));
```
зачем toIntExact? надо просто id сделать типом int и всё

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -69,6 +70,10 @@ public void uploadImage(int id, MultipartFile image) throws IOException {
     }
 
+    public final Optional<AdEntity> get(Long id) {
+        return adRepository.findById(Math.toIntExact(id));
```
зачем toIntExact? надо просто id сделать типом int и всё

Tomokou-ne committed:
Внесены правки:
-изменены методы удаления/редактирования комментариев и объявлений
- изменена строка в UserEntity - изменена строка email на username

Tomokou-ne committed:
Рабочее приложение

Tomokou-ne committed:
Рабочее приложение

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -83,7 +83,7 @@ public ResponseEntity<Comment> addComments(@PathVariable("id") Long id,
             }
     )
-    @PreAuthorize("hasRole('ADMIN')")
+    @PreAuthorize("@commentServiceImpl.getEntity(#id).author.email.equals(#auth.name) or hasRole('ADMIN')")
```
что такое id? такого параметра нет, вы хотя бы проверяете работу приложения?
```suggestion
    @PreAuthorize("@commentServiceImpl.getEntity(#commentId).author.email.equals(#auth.name) or hasRole('ADMIN')")
```

dmitry-bizin replied:
и поле в сущности юзера называется теперь username, а не email

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -104,7 +104,7 @@ public ResponseEntity<Void> deleteComments(@PathVariable("adId") Integer adId,
             }
     )
-    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
+    @PreAuthorize("@commentServiceImpl.getEntity(#adId).author.email.equals(#auth.name) or hasRole('ADMIN')")
```
```suggestion
    @PreAuthorize("@commentServiceImpl.getEntity(#commentId).author.email.equals(#auth.name) or hasRole('ADMIN')")
```
создатель комментария, а не объявления

dmitry-bizin replied:
и поле в сущности юзера называется теперь username, а не email

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/User.java:
```diff
@@ -9,7 +9,7 @@
 public class User {
     private int id;
-    private String email;
+    private String username;
```
здесь поле должно быть называться email, а не username, сейчас вы снова сломали работу с фронтом

Tomokou-ne committed:
Внесены изменения в User

Tomokou-ne committed:
Изменения в CommentController

Tomokou-ne committed:
Удалены неиспользующиеся методы в UserServiceImpl

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -71,9 +84,12 @@ public ResponseEntity<Comment> addComments(@PathVariable("id") Integer id,
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
+    @PreAuthorize("@commentServiceImpl.getEntity(#adId).author.username.equals(#auth.name) or hasRole('ADMIN')")
```
почему используется id объявления-то? Надо достать КОММЕНТАРИЙ по id КОММЕНТАРИЯ, взять у него автора, у автора username и проверить с текущим

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -90,10 +106,12 @@ public ResponseEntity<Void> deleteComments(@PathVariable("adId") Integer adId,
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
+    @PreAuthorize("@commentServiceImpl.getEntity(#adId).author.username.equals(#auth.name) or hasRole('ADMIN')")
```
почему используется id объявления-то? Надо достать КОММЕНТАРИЙ по id КОММЕНТАРИЯ, взять у него автора, у автора username и проверить с текущим

Tomokou-ne committed:
Merge branch 'new_dev' into feature_margo

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/AdsController.java

Tomokou-ne committed:
Полностью изменен auth, добавлены аннотации в entity

Tomokou-ne committed:
Правки по методам регистрации и авторизации пользователя

Tomokou-ne committed:
Изменены названия dto классов, внесены правки в контроллеры, убраны неиспользуемые переменные из контроллеров

Tomokou-ne committed:
Внесены изменения в PreAuthorise deleteComment в классе CommentController

Tomokou-ne committed:
Изменен WebSecurityConfig, добавлены классы с конфигурацией паролей и файлов

Tomokou-ne committed:
Изменен UserEntity, внесены новые строки в БД

Tomokou-ne committed:
Полностью перекроено приложение под UserDetailsManager. Изменены названия основных dto классов, изменены методы работы с entity

Tomokou-ne committed:
Merge branch 'feature_margo' into new_dev

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/CommentController.java
#	src/main/java/ru/skypro/homework/controller/UserController.java
#	src/main/java/ru/skypro/homework/dto/Comments.java
#	src/main/java/ru/skypro/homework/entity/AdEntity.java
#	src/main/java/ru/skypro/homework/entity/CommentEntity.java
#	src/main/java/ru/skypro/homework/entity/UserEntity.java
#	src/main/java/ru/skypro/homework/service/UserService.java
#	src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java
#	src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java
#	src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java
#	src/main/resources/application.properties

Tomokou-ne committed:
Убраны неиспользуемые аннотации к классам

Tomokou-ne committed:
Изменен pom.xml

Tomokou-ne committed:
Merge branch 'feature_margo' into new_dev

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
     <dependency>
-      <groupId>org.postgresql</groupId>
+      <groupId>org.springframework.kafka</groupId>
+      <artifactId>spring-kafka</artifactId>
```
зачем это?

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+    </dependency>
+
+    <dependency>
+      <groupId>junit</groupId>
```
не нужно, уже подключено

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+    </dependency>
+
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
```
не нужно, уже подключено

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+    </dependency>
+
+    <dependency>
+      <groupId>org.testng</groupId>
```
не нужно, уже подключено

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
     <dependency>
-      <groupId>org.liquibase</groupId>
-      <artifactId>liquibase-core</artifactId>
+      <groupId>org.springframework.kafka</groupId>
```
Зачем?

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>testcontainers</artifactId>
```
зачем? я что-то сомневаюсь, что вы будете писать тесты с использованием testcontainers

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>junit-jupiter</artifactId>
```
зачем? я что-то сомневаюсь, что вы будете писать тесты с использованием testcontainers

dmitry-bizin commented on pom.xml:
```diff
@@ -20,56 +20,133 @@
+
+    <dependency>
+      <groupId>org.testcontainers</groupId>
       <artifactId>postgresql</artifactId>
```
зачем? я что-то сомневаюсь, что вы будете писать тесты с использованием testcontainers

Tomokou-ne committed:
Изменены dependencies

dmitry-bizin commented on src/main/java/ru/skypro/homework/HomeworkApplication.java:
```diff
@@ -1,15 +1,14 @@
+import org.springframework.kafka.annotation.EnableKafka;
 
 @SpringBootApplication
+@EnableKafka
```
зачем это??

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/MultipartConfig.java:
```diff
@@ -0,0 +1,25 @@
+import javax.servlet.MultipartConfigElement;
+
+@Configuration
+@EnableAutoConfiguration
+@ComponentScan
```
аннотации не нужны

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/StatisticsServiceConfig.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.Map;
+
+@Configuration
+public class StatisticsServiceConfig {
```
зачем это??

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -1,45 +1,42 @@
-                        .roles(Role.USER.name())
-                        .build();
-        return new InMemoryUserDetailsManager(user);
+    @Autowired
+    public WebSecurityConfig(UserDetailsService userDetailsService) {
+        this.userDetailsService = userDetailsService;
     }
```
не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -1,45 +1,42 @@
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.csrf()
+        http
+                .userDetailsService(userDetailsService)
```
не нужно, спринг сам увидет в контексте

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/ImageController.java:
```diff
@@ -1,30 +1,34 @@
 @RequestMapping("/images")
 public class ImageController {
 
+    @Value("src/main/resources/static")
```
в value надо передавать не значение, а ключ конфигурации, какой смысл там передавать значение-то? сразу присвойте в строку тогда его и всё

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/ImageController.java:
```diff
@@ -1,30 +1,34 @@
+    @Value("src/main/resources/static")
     private String imagesPath;
-    private String adsImagesPath;
+    @Value("/images/user_images")
```
в value надо передавать не значение, а ключ конфигурации, какой смысл там передавать значение-то? сразу присвойте в строку тогда его и всё

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/ImageController.java:
```diff
@@ -1,30 +1,34 @@
-    private String adsImagesPath;
+    @Value("/images/user_images")
     private String usersImagesPath;
+    @Value("/images/ad_images")
```
в value надо передавать не значение, а ключ конфигурации, какой смысл там передавать значение-то? сразу присвойте в строку тогда его и всё

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/ImageController.java:
```diff
@@ -1,30 +1,34 @@
 
     @GetMapping("/users/{inageId}")
-    public byte[] getUserImage(@PathVariable String imageId) throws IOException {
+    public byte[] getUserImage(@PathVariable String imageId, @PathVariable String inageId) throws IOException {
```
здесь неверно, параметр в url 1, а в методе 2

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Comment.java:
```diff
@@ -1,17 +1,21 @@
+@Builder
 public class Comment {
-    private int author;
+    private Long author;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Comment.java:
```diff
@@ -1,17 +1,21 @@
-    private long createdAt;
-    private int pk;
+    private LocalDateTime createdAt;
+    private Long pk;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Comment.java:
```diff
@@ -1,17 +1,21 @@
     private String authorFirstName;
-    private long createdAt;
-    private int pk;
+    private LocalDateTime createdAt;
```
long

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/User.java:
```diff
@@ -1,14 +1,16 @@
+@Builder
 public class User {
-    private int id;
+    private Long id;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/AdEntity.java:
```diff
@@ -1,32 +1,38 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private int pk;
-    @ManyToOne
+    private Long id;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/CommentEntity.java:
```diff
@@ -1,30 +1,33 @@
     @Column(name = "comment_id")
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private int pk;
+    private Long id;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
     @Column(name = "user_id")
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private int id;
+    private Long id;
```
int

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
-    public UserEntity(String password, String username, String firstName, String lastName, String phone, Role role) {
-        this.password = password;
+    @Column(name = "reg_date")
+    private LocalDateTime registrationDate = LocalDateTime.now();
```
не нужно, такого поля нет

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
+    private List<CommentEntity> comments;
+
+    @Column(name = "non_expired")
+    private boolean nonExpired = true;
```
не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
+    private boolean nonCredentialsExpired = true;
+
+    @Column(name = "is_enabled")
+    private boolean isEnabled = true;
```
не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
+    private boolean nonLocked = true;
+
+    @Column(name = "non_credentials_expired")
+    private boolean nonCredentialsExpired = true;
```
не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserEntity.java:
```diff
@@ -1,47 +1,98 @@
+    private boolean nonExpired = true;
+
+    @Column(name = "non_locked")
+    private boolean nonLocked = true;
```
не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserWrapper.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return Collections.singletonList(
+                new SimpleGrantedAuthority(this.userEntity.getRole().toString())
```
неверно, authority должны начинаться с ROLE, т.е. ROLE_ADMIN и ROLE_USER

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserWrapper.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return this.userEntity.isNonExpired();
```
надо просто вернуть true

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserWrapper.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Override
+    public boolean isEnabled() {
+        return this.userEntity.isEnabled();
```
надо просто вернуть true

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserWrapper.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return this.userEntity.isNonCredentialsExpired();
```
надо просто вернуть true

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/UserWrapper.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return this.userEntity.isNonLocked();
```
надо просто вернуть true

dmitry-bizin commented on src/main/java/ru/skypro/homework/mappers/AdMapper.java:
```diff
@@ -1,27 +1,48 @@
-        return new Ads(entity.getAuthor().getId(), entity.getImagePath(),
-                entity.getPk(), entity.getPrice(), entity.getTitle());
+    public Ad entityToAdsDto(AdEntity entity) {
+        int adId = Math.toIntExact(entity.getAuthor().getId());
+        int userId = Math.toIntExact(entity.getId());
```
зачем делать эти касты? ещё раз: id ВЕЗДЕ должны быть типа int

dmitry-bizin commented on src/main/java/ru/skypro/homework/microservice/NestedUserInfo.java:
```diff
@@ -0,0 +1,31 @@
+@Data
+@EqualsAndHashCode
+@NoArgsConstructor
+final class NestedUserInfo implements Serializable {
```
Зачем этот класс? он не нужен

dmitry-bizin commented on src/main/java/ru/skypro/homework/microservice/StatisticsMessage.java:
```diff
@@ -0,0 +1,38 @@
+@Data
+@Slf4j
+@EqualsAndHashCode
+public class StatisticsMessage implements MicroServiceMessage {
```
тоже лишнее

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -25,70 +24,76 @@ public class AdServiceImpl implements AdService {
-        return new ResponseWrapperAds(result.size(), result);
+    @Override
+    public String deleteAd(UserEntity userEntity, Long id) {
+        if (userEntity.getRole().equals(Role.ADMIN)) {
```
зачем сломали прошлые проверки? они были написаны правильно, надо их вернуть

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -25,70 +24,76 @@ public class AdServiceImpl implements AdService {
-    public void delete(int id) {
-        adRepository.deleteById(id);
+    public final AdEntity get(Long id) {
+        return adRepository.findById(id).orElseThrow(FindException::new);
```
комментарии и картинку тоже надо удалять

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/ImageServiceImpl.java:
```diff
@@ -4,49 +4,81 @@
 public class ImageServiceImpl implements ImageService {
-    private final ImageRepository repository;
-    @Value("${path.to.images}")
+    @Value("src/main/resources/static")
```
см. замечание выше про то же самое

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/ImageServiceImpl.java:
```diff
@@ -4,49 +4,81 @@
-    @Value("${path.to.images}")
+    @Value("src/main/resources/static")
     private String imageDirectory;
+    @Value("/images/user_images")
```
см. замечание выше про то же самое

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -2,69 +2,88 @@
     @Override
-    public User get(String name) {
-        return mapper.entityToUserDto(getEntity(name));
+    public UserEntity patch(UserEntity model) {
```
зачем метод из одной строки? зачем дублировать метод? уже есть точно такой же

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -2,69 +2,88 @@
     @Override
-    public User update(User user, String name) {
-        return mapper.entityToUserDto(userRepository.save(mapper.userDtoToEntity(user, getEntity(name))));
+    public UserEntity post(UserEntity model) {
```
зачем метод из одной строки? зачем дублировать метод? уже есть точно такой же

Tomokou-ne committed:
Удалены ненужные dependencies

Tomokou-ne committed:
Изменены классы config

Tomokou-ne committed:
Добавлен путь к изображениям в ImageController

Tomokou-ne committed:
Изменен параметр в GET getUserImage

Tomokou-ne committed:
Внесены изменения в типы переменных dto comment & user

Tomokou-ne committed:
Изменены типы id в entity

Tomokou-ne committed:
Внесены изменения в mappers

Tomokou-ne committed:
Удален микросервис

Tomokou-ne committed:
Внесены изменения в service

Tomokou-ne committed:
Изменены параметры переменных в сервис классах ==> в классах контроллерах.
Улучшен метод удаления объявления.
Изменен метод getAuthorities в классе UserWrapper

Tomokou-ne committed:
Добавлен метод конвертации localdatetime в CommentMapper

Tomokou-ne committed:
Добавлены изображения

Tomokou-ne committed:
Изменен wrapper и create.sql

Tomokou-ne committed:
Изменен метод: loadByUsername

Tomokou-ne committed:
Внесены изменения в AdMapper, изменена версия сприг с 3.0.4 на 2.9.14(?)

Tomokou-ne committed:
Внесены изменения в AdMapper

Tomokou-ne committed:
Добавлены аннотации в dto классы

Tomokou-ne committed:
Изменен метод добавления и обновления комментариев

Tomokou-ne committed:
Добавлены проверки на удаление/изменение комментариев и объявлений
